csrc/sign.cpp

```diff
@@ -76,6 +77,32 @@ bool initializeContext(raii_env& env,
             throw_openssl("Unable to create MD_CTX");
         }
 
+#if !defined(FIPS_BUILD) || defined(EXPERIMENTAL_FIPS_BUILD)
+        if (preHash && EVP_PKEY_id(pKey) == EVP_PKEY_ED25519) {
```

Comment:
Since you plan on having other pre-hash algorithms. It makes sense to me for you to pull down this `pkey==ED25519` logic to nest underneath the `if(preHash)`.

Reply:
This block of logic is very much specific to Ed25519 and the NID conversion we do there. We do have other `preHash` (the local variable) above on L74 for ML-DSA ExternalMu, and it applies differently.